internal/pkg/manifest/backend_svc.go

```diff
@@ -101,37 +102,38 @@ func (s *BackendService) MarshalBinary() ([]byte, error) {
 
 // DockerfilePath returns the image build path.
 func (s *BackendService) DockerfilePath() string {
-	return s.Image.Build
+	return aws.StringValue(s.Image.Build)
 }
 
 // ApplyEnv returns the service manifest with environment overrides.
 // If the environment passed in does not have any overrides then it returns itself.
-func (s *BackendService) ApplyEnv(envName string) *BackendService {
-	target, ok := s.Environments[envName]
+func (s BackendService) ApplyEnv(envName string) (*BackendService, error) {
+	overrideConfig, ok := s.Environments[envName]
 	if !ok {
-		return s
+		return &s, nil
 	}
-	return &BackendService{
-		Service: s.Service,
-		Image: imageWithPortAndHealthcheck{
-			ServiceImageWithPort: target.Image.ServiceImageWithPort,
-			HealthCheck:          target.Image.HealthCheck,
-		},
-		TaskConfig: s.TaskConfig.copyAndApply(target.TaskConfig),
-		Sidecar:    s.Sidecar.copyAndApply(target.Sidecar),
-		LogConfig:  s.LogConfig.copyAndApply(target.LogConfig),
+	target := BackendService{
+		Image:      overrideConfig.Image,
+		TaskConfig: overrideConfig.TaskConfig,
+		LogConfig:  overrideConfig.LogConfig,
+		Sidecar:    overrideConfig.Sidecar,
+	}
+	if err := mergo.Merge(&s, target, mergo.WithOverride, mergo.WithOverwriteWithEmptyValue); err != nil {
```

Comment:
I wonder if once target is merged it will leave the original `*s` to be the same as before the overrides.

Can we add a test for that, say that we have the default value set to `cpu=256` and add an env override with `cpu=512`. Besides checking that the output is `512` can we check if the original remains `256`?

Reply:
The original `s` won't change since I don't use a pointer of `BackendService` as the receiver. But yeah for safe i'll add a test for that.

Reply:
Added!